src/libAtomVM/memory.h

```diff
@@ -250,6 +251,15 @@ MUST_CHECK static inline enum MemoryGCResult memory_ensure_free(Context *ctx, si
     return memory_ensure_free_opt(ctx, size, MEMORY_NO_SHRINK);
 }
 
+/**
+ * @brief makes sure that the given nif environment has given free memory
+ *
+ * @details this function makes sure that at least size terms are available. When not available, a new fragment will be allocated.
+ * @param env the target environment.
+ * @param size needed available memory.
+ */
+MUST_CHECK enum MemoryGCResult memory_erl_nif_env_ensure_free(ErlNifEnv *env, size_t size);
```

Comment:
so, if I understood correctly this function is an addition over the original API.
So, if I understood correctly, our `enif_*` functions mimic quite well the original API except that this function needs to be called otherwise a crash will happen, right?

Reply:
This function is exclusively used in `enif_make_resource` for now and is meant to be used in other `enif_make_*` functions. The idea is to mimic as much as possible the original API and this function should not be used by drivers.